### PR TITLE
Auth v2

### PR DESCRIPTION
**Pull request by hpx7**

### CODE REVIEW

**Comment by gwprice115** [comment]
maybe I'm missing something but it seems like if i login anonymously i can't choose my username?

**Comment by hpx7** [comment]
yeah @gwprice115 if you use anonymous login want you want a session-scoped username, you'll want to implement it as part of your app logic (e.g. having a `joinGame` method which takes a `nickname` input). does that make sense?

**Comment by gwprice115** [comment]
oh yeah right

File: examples/avalon/impl.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import { Methods } from "./.rtag/methods";
 import {
-  PlayerData,
+  UserData,
```
Comment (knigam) [inline examples/avalon/impl.ts:3]: how is UserData defined? I don't see this in types.yml anywhere

File: templates/base/package.json.hbs [hunk 2]
```diff
@@ -4,11 +4,14 @@
     "@blueprintjs/icons": "3.24.0",
     "axios": "0.21.1",
     "express": "4.17.1",
+    "google-auth-library": "6.1.6",
```
Comment (knigam) [inline templates/base/package.json.hbs:7]: Do you need to always import these here? Can't they be added depending on which auth methods are actually chosen?

File: examples/avalon/impl.ts (hunk 1 above)
Comment (knigam) [reply examples/avalon/impl.ts:3]: Nevermind, I see this in types.ts

**Review by knigam — APPROVED** [review]

File: templates/base/types.ts.hbs [hunk 3]
```diff
@@ -28,6 +28,22 @@ export interface {{makeRequestName @key}} {
 {{/if}}
 }
 {{/each}}
+{{#each auth}}
+export interface {{capitalize @key}}UserData {
+  type: "{{@key}}";
+  id: string;
```
Comment (knigam) [inline templates/base/types.ts.hbs:34]: in other files you seem to assume that "username" exists in UserData. You should either have these {Auth}UserData interfaces implement an interface that contains "type", "id", and "username" or at the very least ensure "username" exists outside of the if blocks below.

**Pull request merged**